lib/kickplan/adapters/http.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Kickplan
+  module Adapters
+    class HTTP < Adapter
+      extend Forwardable
+
+      delegate %i(get post put delete) => :connection
+
+      def connection
+        memoize do
+          Faraday::Connection.new(
+            url: config.endpoint,
+            proxy: config.proxy,
+            builder: config.middleware,
+            headers: {
+              user_agent: config.user_agent
+            }
+          )
+        end
+      end
+
+      def resolve(key = nil, params = {})
+        if key.nil? || key.is_a?(Hash)
+          key, params = nil, key
+        end
+
+        post("features/#{key}", params).body
+      end
+    end
+  end
+
+  Adapters.register(:http, Adapters::HTTP)
```

Comment:
This is the adapter registration api. This will allow sdk users to define their own adapters and register them. They can be configured via:

```
Kickplan.configure do |config|
  config.adapter = :http
  ...
end
```